### PR TITLE
docs: cleanup changelog

### PR DESCRIPTION
**Pull request by mpoke**
## Description

Closes: NA

Cleanup changelog. 

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct `docs:` prefix in the PR title
- [x] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [x] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] Confirmed the correct `docs:` prefix in the PR title
- [ ] Confirmed all author checklist items have been addressed 
- [ ] Confirmed that this PR only changes documentation
- [ ] Reviewed content for consistency
- [ ] Reviewed content for thoroughness
- [ ] Reviewed content for spelling and grammar
- [ ] Tested instructions (if applicable)

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -34,73 +34,20 @@ Ref: https://keepachangelog.com/en/1.0.0/
...
+## [Unreleased]
 
-* (chore) Update module version to v10 [#2502](https://github.com/cosmos/gaia/pull/2502).
-* (deps) Bump multiple dependencies (sdk, cometbft, ibc-go, btcd), see [#2498](https://github.com/cosmos/gaia/pull/2498) for details.
-  * (deps) Bump [cosmos-sdk](https://github.com/cosmos/cosmos-sdk) to [v0.45.16-ics](https://github.com/cosmos/cosmos-sdk/releases/tag/v0.45.16-ics). See [CHANGELOG.md](https://github.com/cosmos/cosmos-sdk/blob/release/v0.45.x/CHANGELOG.md) for details.
-  * (deps) Bump [ibc-go](https://github.com/cosmos/ibc-go) to [v4.4.0](https://github.com/cosmos/ibc-go/blob/release/v4.4.x/CHANGELOG.md). See [v4.2 Release Notes](https://github.com/cosmos/ibc-go/releases/tag/v4.4.0) for details.
-* (gaia) Bump Golang prerequisite from 1.18 to 1.20 see (https://go.dev/blog/go1.20) for details.
+## [v10.0.0] 2023-05-19
 
-## [v9.1.0] - 2023-05-08
```
Comment (shaspitz) [inline CHANGELOG.md:45]: This release branch is based off v9.1.0, wouldn't it make sense for changelog readers to have context on the version this branch is built off?

**Review by shaspitz — APPROVED** [review]
just one comment, I'll still give approval

File: CHANGELOG.md (hunk 1 above)
Comment (mpoke) [reply CHANGELOG.md:45]: I think that every new major release would be based off the latest release, right? 

Also, to simplify the changelog of a release branch, we should just keep the versions related to that release branch.

**Review by MSalopek — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (shaspitz) [reply CHANGELOG.md:45]: > I think that every new major release would be based off the latest release, right?

No typically a new major release branch is based off main. The v10 release branch deviates from main (it's based off v9.1.0), and that's why I think the v9 entries are relevant

Comment (shaspitz) [reply CHANGELOG.md:45]: > we should just keep the versions related to that release branch.

Agreed, and in that case the v9 entries are relevant imo

**Pull request merged**